src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+    private int capacity;
 
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        capacity = DEFAULT_CAPACITY;
+        threshold = (int) (capacity * LOAD_FACTOR);
+    }
+    
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        int hash = getHash(key);
+        Node<K, V> newNode = new Node<>(key, value);
+        Node<K, V> previousNode = table[hash % capacity];
+        if (previousNode == null) {
+            table[hash % capacity] = newNode;
+            size++;
+            return;
+        }
+        while (previousNode != null) {
+            if (key == previousNode.key || key != null && key.equals(previousNode.key)) {
+                previousNode.value = value;
+                return;
+            }
+            if (previousNode.next == null) {
+                previousNode.next = newNode;
+                size++;
+                return;
+            }
+            previousNode = previousNode.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> getNode = table[getHash(key) % capacity];
+        while (getNode != null) {
+            if (key == getNode.key || key != null && key.equals(getNode.key)) {
```

Comment:
Here we are already allowed to use Objects.equals, so it is better to use this method